pkg/exemplars/exemplars.go

```diff
@@ -79,15 +79,11 @@ func (rr *GRPCClient) Exemplars(ctx context.Context, req *exemplarspb.ExemplarsR
 		return nil, nil, errors.Wrap(err, "proxy Exemplars")
 	}
 
-	resp.data = dedupExemplarsData(resp.data, rr.replicaLabels)
-	for _, d := range resp.data {
-		d.Exemplars = dedupExemplars(d.Exemplars, rr.replicaLabels)
-	}
-
+	resp.data = dedupExemplarsSeriesLabels(resp.data, rr.replicaLabels)
```

Comment:
I'm a bit confused. You say:

> This pr removes the step of adding replica labels to exemplars. Only adding replica labels to exemplars' series labels.

`dedupExemplarsSeriesLabels` deduplicates the data using the replica labels and removes those labels. Here now you are now only doing it on the series so series will be without the replica labels, and exemplars - will be with the replica labels. Shouldn't it be the other way around i.e. the deduplication should only happen on the exemplars? :smile:

Reply:
Sorry, that was my big mistake. Thanks for catching it. I should modify the code from the Prometheus proxy side.

Reply:
Updated. PTAL